crates/subspace-gateway/src/commands/run/dsn.rs

```diff
@@ -1,26 +1,38 @@
 //! DSN config and implementation for the gateway.
 
-use clap::Parser;
+use crate::node_client::{NodeClient, RpcNodeClient};
+use anyhow::anyhow;
+use clap::{Parser, ValueHint};
 use subspace_networking::libp2p::kad::Mode;
 use subspace_networking::libp2p::Multiaddr;
 use subspace_networking::{construct, Config, KademliaMode, Node, NodeRunner};
+use tracing::{debug, info};
 
 /// Configuration for network stack
 #[derive(Debug, Parser)]
 pub(crate) struct NetworkArgs {
-    /// Multiaddrs of DSN bootstrap nodes to connect to on startup, multiple are supported
+    /// WebSocket RPC URL of the Subspace node to connect to.
+    ///
+    /// This node provides the DSN protocol version, default bootstrap nodes, and piece validation
+    /// metadata.
+    #[arg(long, value_hint = ValueHint::Url, default_value = "ws://127.0.0.1:9944")]
+    node_rpc_url: String,
+
+    /// Multiaddrs of DSN bootstrap nodes to connect to on startup, multiple are supported.
+    ///
+    /// The default bootstrap nodes are fetched from the node RPC connection.
     #[arg(long)]
     pub(crate) dsn_bootstrap_nodes: Vec<Multiaddr>,
 
+    /// Multiaddrs of DSN reserved nodes to maintain a connection to, multiple are supported.
+    #[arg(long)]
+    dsn_reserved_peers: Vec<Multiaddr>,
+
     /// Enable non-global (private, shared, loopback..) addresses in the Kademlia DHT.
     /// By default these addresses are excluded from the DHT.
     #[arg(long, default_value_t = false)]
     pub(crate) allow_private_ips: bool,
```

Comment:
This field needs to be `pub(crate)` or `pub(super)` for `--dev`, but I'd hide the rest as you did with reserved peers.